### PR TITLE
Initial work replace RunmeLauncherProvider

### PR DESCRIPTION
**Pull request by pastuxso**
This approach is looking forward to re-using the same source to provide a list of Tasks. Now is grouping cells by Notebook, and they are opening the Notebook in the specific cell. This is the first iteration of future improvements on it.

To activate it, the following setting is required: `"runme.features.NewTreeProvider": true`

image

### CODE REVIEW

**Comment by sourishkrout** [comment]
where's the icon @pastuxso?

image

**Comment by pastuxso** [comment]
@sourishkrout Icon added. I don't know why it was ignored.

File: src/extension/provider/launcherBeta.ts [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+    )
+
+    this.ready = new Promise((resolve) => {
+      const disposable = server.onTransportReady(() => {
+        disposable.dispose()
+        resolve()
+      })
+    })
+
+    this.tasks = lastValueFrom(this.loadProjectTasks())
```
Comment (sourishkrout) [inline src/extension/provider/launcherBeta.ts:92]: why duplicate the work the task loader's already done?

File: src/extension/provider/launcherBeta.ts [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+/**
+ * used to force VS Code update the tree view when user expands/collapses all
+ * see https://github.com/microsoft/vscode/issues/172479
+ */
+/* eslint-disable-next-line */
+let sauceCount = 0
+
+export class RunmeLauncherProvider implements RunmeTreeProvider {
+  #disposables: Disposable[] = []
+  private allowUnnamed = true
```
Comment (sourishkrout) [inline src/extension/provider/launcherBeta.ts:64]: Unnamed tasks clutter. Why not `false` like the old default?

File: src/extension/provider/launcherBeta.ts [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+          mdBuffer = await workspace.fs.readFile(Uri.parse(documentPath))
+        } catch (err: any) {
+          if (err.code !== 'FileNotFound') {
+            logger.error(`${err.message}`)
+          }
+          throw err
+        }
+
+        const token = new CancellationTokenSource().token
+        notebook = from(this.serializer.deserializeNotebook(mdBuffer, token))
```
Comment (sourishkrout) [inline src/extension/provider/launcherBeta.ts:229]: Again, why duplicate the work the task loader's already done? it's expensive to traverse plus deserialize files in a project/repo multiple times

File: src/extension/provider/launcherBeta.ts (hunk 1 above)
Comment (sourishkrout) [reply src/extension/provider/launcherBeta.ts:92]: isn't this what the vscode API `tasks.fetchTasks({ type: 'runme' })` is for?

**Review by sourishkrout — CHANGES_REQUESTED** [review]
The unnamed tasks visibility default must match the legacy launcher to merge this first version.

The performance impacts of not using the vscode APIs can be addressed in separate PRs—however, there will be no public release until these are solved.

**Comment by sourishkrout** [comment]
I am merging this with the correct default. Please address the remaining comments in separate issues/tickets when we pick this up again, @pastuxso.

**Comment by pastuxso** [comment]
@sourishkrout no problem. Go ahead.

**Comment by pastuxso** [comment]
Work in progress https://github.com/stateful/vscode-runme/pull/1784

**Pull request merged**